### DISCUSSION TITLE
Export tracks

### DISCUSSION THREAD

**Discussion by asmir19** [Karry/osmscout-sailfish#292] [Q&A] [answered]
![Screenshot_20220314_001](https://user-images.githubusercontent.com/77565818/158118246-1f9bb194-1345-4eb2-9bc6-32f465d08c26.png)
![Screenshot_20220314_002](https://user-images.githubusercontent.com/77565818/158118264-9a32f851-cd29-4a11-8864-849cd8330bd1.png)
![Screenshot_20220314_003](https://user-images.githubusercontent.com/77565818/158118287-d87b9774-572e-405d-981d-9484d100fca8.png)
Hi, Karry!
Much thanks for OSM Scout! App (v.2.22-1) works fine: maps, navigation and track recording. But there is one little bug: viewing track's details and exporting to the file are not working.
Please, comment on the situation and tell me what I'm doing wrong.
Best regards, Asmir

**Comment by Karry** [comment, marked as answer]
Hi. It is really strange. What device you are using? Do you have latest Sailfish OS release? If you have enabled root access to your phone, can you execute `journalctl -xe | grep osmscout` and post output of this command? Or just as a normal user execute `/usr/bin/sailjail -- /usr/bin/harbour-osmscout` in terminal and post logs here. 

Thank you.

**Reply by asmir19** [reply]
Thank you for response!
I'm using Inoi R7, OS release is 4.1.024.

   **Log:**
[nemo@Sailfish ~]$ /usr/bin/sailjail -- /usr/bin/harbour-osmscout
Error: dumpable process
constructing /run/firejail/mnt/privileged:  ...
mounting /run/firejail/mnt/privileged @ /home/nemo/.local/share/system/privileged
hiding /run/firejail/mnt/privileged
Starting harbour-osmscout 2.22.0 (libosmscout ad0e5bf84, Qt 5.6.3, Unknown OS)
[D] main:147 - Usage of memory mapped files is supported.
[D] Migration::migrate:78 - Considering migration "/home/nemo/.config/harbour-osmscout/harbour-osmscout.conf" to "/home/nemo/.config/cz.karry.osmscout/OSMScout/OSMScout.conf"
[D] Migration::migrate:78 - Considering migration "/home/nemo/.config/cz.karry.osmscout/OSMScout.conf" to "/home/nemo/.config/cz.karry.osmscout/OSMScout/OSMScout.conf"
[D] Migration::migrate:78 - Considering migration "/home/nemo/.local/share/harbour-osmscout/harbour-osmscout" to "/home/nemo/.local/share/cz.karry.osmscout/OSMScout"
[D] Migration::wipeOldCache:121 - Considering cache wipe "/home/nemo/.cache/harbour-osmscout/harbour-osmscout"
[D] Migration::migrate:78 - Considering migration "/home/nemo/Maps" to "/home/nemo/Downloads/Maps"
[D] Migration::migrate:78 - Considering migration "/home/nemo/Documents/Maps" to "/home/nemo/Downloads/Maps"
[W] main:251 - Can't load translator for locale QLocale(English, Latin, UnitedStates) / "en_US" ( "/usr/share/harbour-osmscout/translations" )
[D] AppSettings::settingFile:422 - config: "/home/nemo/.config/cz.karry.osmscout/OSMScout/OSMScout.conf"
[W] osmscout::Settings::loadOnlineTileProviders:138 - Couldn't open "/usr/share/harbour-osmscout/resources/online-tile-providers.json" file.
[W] osmscout::{anonymous}::loadResourceProviders:175 - Couldn't open "/usr/share/harbour-osmscout/resources/map-providers.json" file.
[W] osmscout::{anonymous}::loadResourceProviders:175 - Couldn't open "/usr/share/harbour-osmscout/resources/voice-providers.json" file.
[D] osmscout::MapManager::MapManager:236 - MapManager ctor
[D] MemoryManager::memoryLevelChanged:90 - Memory level: "normal"
[D] osmscout::DBThread::onDatabaseListChanged:270 - Basemap found and loaded from ' "/usr/share/harbour-osmscout/resources/world" '...
[D] Storage::init:630 - Storage database opened: "/home/nemo/.local/share/cz.karry.osmscout/OSMScout/storage.db"
[D] Storage::updateSchema:501 - creating collection table
[D] Storage::updateSchema:512 - creating track table
[D] Storage::updateSchema:523 - creating track_segment table
[D] Storage::updateSchema:534 - creating track_point table
[D] Storage::updateSchema:545 - creating waypoints table
[D] MemoryManager::memoryLevelChanged:90 - Memory level: "unknown"
[D] Storage::updateSchema:556 - creating search_history table
[W] MemoryManager::memoryLevelChanged:105 - Unsupported Memory level: "unknown"
[D] Storage::updateSchema:574 - creating idx_track_point_segment_id index
[D] unknown:0 - Using Wayland-EGL
[W] unknown:-1 - file:///usr/share/harbour-osmscout/qml/main.qml: File not found
library "/vendor/lib/egl/libGLESv2S3D_adreno.so" not found

_By the way, during track's recording you can monitor current values such as distance, speed, altitude, etc..._

**Reply by Karry** [reply]
these errors are strange
```
Couldn't open "/usr/share/harbour-osmscout/resources/online-tile-providers.json
Couldn't open "/usr/share/harbour-osmscout/resources/map-providers.json" file.
```
And with this line:
```
 file:///usr/share/harbour-osmscout/qml/main.qml: File not found
```
I am surprised that main map is working for you... Can you check that these files exists on your phone? As app was able to load basemap from `/usr/share/harbour-osmscout/resources/world`, at least some resources are on correct place...

**Reply by asmir19** [reply]
Hi, Karry!
It's very strange, cause all files are present:

_[root@Sailfish nemo]# cd /usr/share/harbour-osmscout/resources/
[root@Sailfish resources]# ls -a
.                           online-tile-providers.json
..                          voice-providers.json
map-providers.json          world
[root@Sailfish resources]# cd world
[root@Sailfish world]# ls -a
.          ..         water.idx
[root@Sailfish world]# cd /usr/share/harbour-osmscout/qml
[root@Sailfish qml]# ls -a
.                    SpeedIndicator.qmlc  main.qml
..                   custom               pages
AboutDialog.qml      desktop.qmlc         qmldir
SearchDialog.qml     l10n.qml_

And one lib (**libGLESv2S3D_adreno.so**) is absent:

_[root@Sailfish qml]# cd /vendor/lib/egl/
[root@Sailfish egl]# ls -a
.                       libEGL_adreno.so        libq3dtools_adreno.so
..                      libGLESv1_CM_adreno.so
eglsubAndroid.so        libGLESv2_adreno.so_

**Reply by Karry** [reply]
Can you try to run it without Sailjail please? Just start `/usr/bin/harbour-osmscout`...

**Reply by asmir19** [reply]
I did it. I ran osmscout and tried to open one track. I've got the same result: "could not load page". Log-file is attached.
[osmscout_20220315.log](https://github.com/Karry/osmscout-sailfish/files/8252865/osmscout_20220315.log)

I hope, it will be usefull.

**Reply by Karry** [reply]
so, the main issue is:
```
[W] unknown:1283 - file:///usr/lib/qt5/qml/Sailfish/Silica/PageStack.qml:1283:13: QML AnimatedLoader: (file:///usr/share/harbour-osmscout/qml/pages/Collection.qml:23:1: module "Sailfish.Share" is not installed
    import Sailfish.Share 1.0
    ^)
```
the [new sharing api](https://sailfishos.org/develop/docs/declarative-transferengine/qml-sailfishshare-sailfish-share-shareaction.html/) was introduced in [Sailfish OS 4.2.0 "Verla"](https://forum.sailfishos.org/t/release-notes-verla-4-2-0/7092). And from the log, you have `Sailfish Mobile OS RUS 4.1.0.24+omp0.1.3.25 (Kvarken)`. Is there any possibility to upgrade to 4.2 or migrate to Sailfish OS from Jolla? If not, I may suggest you to uninstall OSM Scout and install [OSM Scout for Jolla 1 ](https://openrepos.net/content/karry/osm-scout-jolla-1) that is the same as OSM Scout in Harbour, just commits with new sharing api are dropped...

**Reply by asmir19** [reply]
Oh, much thanks, Karry! Version 4.1 is more usable than 4.2 (to my opinion). I'll try version for Jolla 1.
Best regards, Asmir
